sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/GenericTypeParameterUnused.cs

```diff
@@ -65,7 +66,10 @@ protected override void Initialize(SonarAnalysisContext context)
                             return;
                         }
 
-                        var helper = GetTypeParameterHelper(methodDeclaration, classDeclaration);
+                        // For local methods the only valid modifiers are async, static and unsafe
+                        // https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions#local-function-syntax
+
+                        var helper = GetTypeParameterHelper(methodDeclaration, classDeclaration, localFunctionDeclaration);
```

Comment:
maybe rename `helper` to `typeParameterHelper` or `typeParameterInformation` ?

Reply:
Yes, the naming (already existing) is really unfortunate. I didn't want to change it to ease the code review.